lib/rules/utils/ast-utils.js

```diff
@@ -27,8 +27,8 @@ const {
 const anyFunctionPattern = /^(?:Function(?:Declaration|Expression)|ArrowFunctionExpression)$/u;
 const anyLoopPattern = /^(?:DoWhile|For|ForIn|ForOf|While)Statement$/u;
 const arrayOrTypedArrayPattern = /Array$/u;
-const arrayMethodPattern = /^(?:every|filter|find|findIndex|forEach|map|some)$/u;
 const bindOrCallOrApplyPattern = /^(?:bind|call|apply)$/u;
+const methodWithThisArgPattern = /^(?:every|filter|find(?:Last)?(?:Index)?|flatMap|forEach|map|some)$/u;
```

Comment:
Should we keep "array" in the name?

```suggestion
const arrayMethodWithThisArgPattern = /^(?:every|filter|find(?:Last)?(?:Index)?|flatMap|forEach|map|some)$/u;
```

Reply:
I think it's fine to keep `array` since these are all names of array methods currently. We will notice if the situation will change. Updated in https://github.com/eslint/eslint/pull/17439/commits/01dfe5aa4363e6003a2b7beacb9e533116923e06.